apps/site/lib/site_web/views/event_view.ex

```diff
@@ -83,4 +83,13 @@ defmodule SiteWeb.EventView do
       pretty_date(end_time, "{WDshort}, {Mshort} {D}, {YYYY}")
     } #{format_time(end_time)}"
   end
+
+  @spec date_upcoming(%{
+          start: NaiveDateTime.t() | DateTime.t(),
+          stop: NaiveDateTime.t() | DateTime.t()
+        }) :: boolean
+  def date_upcoming(range) do
+    (!!range.stop and Date.compare(range.stop, Timex.now("America/New_York")) == :gt) or
+      (!range.stop and Date.compare(range.start, Timex.now("America/New_York")) == :gt)
+  end
```

Comment:
Oh, it took me a while to figure this out, but DateTime.compare doesn't allow NaiveDateTime and DateTime.  I like both types equally ;) so I guess I'll just pick one type to keep and one type to convert

Reply:
I like the use of `Date.compare` here! I'd like to see a test for this function.

I'm also curious if using pattern matching and guards to check if the dates exist would be helpful to simplify the logic. Something like 

```elixir
# check the start/stop
def date_upcoming((%{
           start: start,
           stop: stop
         }) when not is_nil(start) and not is_nil(stop) do
  # date comparing!
end

# everything else
def date_upcoming(_), do: false

```

Reply:
I realized a little later that using `Date.compare` means we couldn't filter events by what time the event occurred.  I'd prefer DateTime.compare so we can hide expired events that happened earlier in the day.  What do you think?

Reply:
What is totally stumping me though:  I need to import Tzdata.TimeZoneDatabase into `util.ex` so that I can convert NaiveDateTimes into DateTimes before comparing them.  But I cannot figure out how to import this!

Reply:
For that conversion, can you check if `Util.convert_using_timezone` does what you need?

Reply:
Maybe it'd make sense to check the date type first? And if it's a NaiveDateTime use NaiveDateTime.compare, otherwise use DateTime.compare (which I do think makes sense to use so you can hide events by time)? Not sure if you still need a time zone in that case though..hm